kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventConsumer.java

```diff
@@ -0,0 +1,117 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValue;
+import io.micrometer.core.instrument.Counter;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+
+@Slf4j
+public class KafkaEventConsumer<K, V, D, C> extends Thread {
+  private static final String EVENT_CONSUMER_ERROR_COUNT = "event.consumer.error.count";
+  private static final String TOPIC_NAME = "topic.name";
+  private static final String POLL_TIMEOUT = "poll.timeout";
+  private final List<TopicPartition> topicPartitions;
+  private final KafkaConsumer<D, C> kafkaEventConsumer;
+  private final Duration pollTimeout;
+  private final Counter errorCounter;
+  private final KafkaEventListener<K, V> kafkaEventListener;
+
+  public KafkaEventConsumer(
+      String consumerName,
+      Config kafkaConfig,
+      Deserializer<D> keyDeserializer,
+      Deserializer<C> valueDeserializer,
+      KafkaEventListener<K, V> kafkaEventListener) {
```

Comment:
The consumer's constructor requires the listener and the listener constructor takes a list of consumers? Can we make this relationship one (either) way?

Reply:
Oops, my bad. Fixing